client/asset/btc/btc.go

```diff
@@ -5015,6 +5015,117 @@ func (btc *baseWallet) RefundBond(ctx context.Context, ver uint16, coinID, scrip
+		}
+		if !bytes.Equal(pkScript, msgTx.TxOut[0].PkScript) {
+			return nil, fmt.Errorf("bond script does not match commit data for %s: %x != %x",
+				txHash, bondScript, msgTx.TxOut[0].PkScript)
+		}
+		return &asset.BondDetails{
+			Bond: &asset.Bond{
+				Version: 0,
+				AssetID: BipID,
+				Amount:  uint64(msgTx.TxOut[0].Value),
+				CoinID:  coinID,
+				Data:    bondScript,
+				//
+				// SignedTx and UnsignedTx not populated because this is
+				// an already posted bond and these fields are no longer used.
+				// SignedTx, UnsignedTx []byte
+				//
+				// RedeemTx cannot be populated because we do not have
+				// the private key that only core knows. Core will need
+				// the BondPKH to determine what the private key was.
+				// RedeemTx []byte
+			},
+			LockTime: time.Unix(int64(lockTime), 0),
+			AcctID:   acct,
+			BondPKH:  pkh[:],
+			PKHFn: func(bondKey *secp256k1.PrivateKey) []byte {
+				pk := bondKey.PubKey().SerializeCompressed()
+				return btcutil.Hash160(pk)
+			},
+		}, nil
+	}
+
+	// If the bond was funded by this wallet or had a change output paying
+	// to this wallet, it should be found here.
+	tx, err := btc.node.getWalletTransaction(txHash)
+	if err == nil {
+		msgTx, err := btc.deserializeTx(tx.Bytes)
+		if err != nil {
+			return nil, fmt.Errorf("invalid hex for tx %s: %v", txHash, err)
+		}
+		return decodeV0BondTx(msgTx)
+	}
+	if !errors.Is(err, asset.CoinNotFoundError) {
+		btc.log.Warnf("Unexpected error looking up bond output %v:%d", txHash, vout)
+	}
+
+	// The bond was not funded by this wallet or had no change output when
+	// restored from seed. This is not a problem. However, we are unable to
+	// use filters because we don't know any output scripts. Brute force
+	// finding the transaction.
+	bestBlockHdr, err := btc.node.getBestBlockHeader()
+	if err != nil {
+		return nil, fmt.Errorf("unable to get best hash: %v", err)
+	}
+	blockHash, err := chainhash.NewHashFromStr(bestBlockHdr.Hash)
+	if err != nil {
+		return nil, fmt.Errorf("invalid best block hash from %s node: %v", btc.symbol, err)
+	}
+	var (
+		blk         *wire.MsgBlock
+		msgTx       *wire.MsgTx
+		searchUntil = time.Now().Add(-365 * 24 * time.Hour) // long!
+	)
+out:
+	for {
+		blk, err = btc.tipRedeemer.getBlock(*blockHash)
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving block %s: %w", blockHash, err)
+		}
+		if blk.Header.Timestamp.Before(searchUntil) {
+			return nil, fmt.Errorf("searched blocks until %v but did not find the bond tx %s", searchUntil, txHash)
+		}
+		for _, tx := range blk.Transactions {
+			if tx.TxHash() == *txHash {
+				btc.log.Debugf("Found mined tx %s in block %s.", txHash, blk.BlockHash())
+				msgTx = tx
+				break out
+			}
+		}
+		blockHash = &blk.Header.PrevBlock
+	}
+	if msgTx == nil {
+		// NOTE: Server made us look for a bogus transaction, we should
+		// be furious.
+		return nil, fmt.Errorf("did not find the bond tx %s", txHash)
```

Comment:
This will never get hit because the only time you break out of the loop is when you find it.

Reply:
Yeah I guess not.

Reply:
Removed this check but added one for a nil hash that also may not be possible.